JavaScript package template fails to build on Linux

Unfortunately as a result of [this fix](https://github.com/DeterminateSystems/zero-to-nix/pull/174) I’m now seeing this error from `nix build`:

```
npm ERR! [esbuild] Failed to find package "esbuild-linux-64" on the file system
```

I think that this is because package-lock.json was generated on a macOS system so it only includes `esbuild-darwin-arm64` and not `esbuild-linux-64` (or other architectures).

--- a/nix/templates/pkg/javascript/flake.nix
+++ b/nix/templates/pkg/javascript/flake.nix
@@ -19,13 +19,13 @@
         default = pkgs.buildNpmPackage {
           name = "zero-to-nix-javascript";
 
-          buildInputs = [
-            pkgs.nodejs-18_x
+          buildInputs = with pkgs; [
+            nodejs-18_x
           ];
 
           src = ./.;
 
-          npmDepsHash = "sha256-URiXuRG7FsQ9tL4DSOxDjG+V+FSWxHnfzi0wvZL+pJA=";
+          npmDepsHash = "sha256-A85l8kFgIU2grgDQNBM7ilLVPehMl6ilkpt4YoiZyeo=";
 
           installPhase = ''
             mkdir $out

